bigquery/rangevalue.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bigquery
+
+// RangeValue represents a continuous RANGE of values of a given element
+// type.  The supported element types for RANGE are currently the BigQuery
+// TIME, TIMESTAMP, and DATE types.
+type RangeValue struct {
```

Comment:
can we add a `Type FieldType` field here to show if the RangeValue is a `TIME`, `TIMESTAMP` or `DATE` here ? This way the `inferSchema` method can also fully infer the type.

Reply:
This doesn't actually help things.  Infer consumes the go type information, it doesn't look at the current values.  In order to make this tractable you'd need N versions of the type, one for each element type.  That didn't feel like a great choice either.

Reply:
Closing this out as we're missing value information for inference entirely.